File: Tracks/csharp/house/House.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class House
{
    private static readonly SortedDictionary<int, string[]> nameAction = new SortedDictionary<int, string[]>
    {
        { 1, new string[2] { "house", "Jack built" } },
        { 2, new string[2] { "malt", "lay in" } },
        { 3, new string[2] { "rat", "ate" } },
        { 4, new string[2] { "cat", "killed" } },
        { 5, new string[2] { "dog", "worried" } },
        { 6, new string[2] { "cow with the crumpled horn", "tossed" } },
        { 7, new string[2] { "maiden all forlorn", "milked" } },
        { 8, new string[2] { "man all tattered and torn", "kissed" } },
        { 9, new string[2] { "priest all shaven and shorn", "married" } },
        { 10, new string[2] { "rooster that crowed in the morn", "woke" } },
        { 11, new string[2] { "farmer sowing his corn", "kept" } },
        { 12, new string[2] { "horse and the hound and the horn", "belonged to" } }
    };

    public static string Recite(int verseNumber) => "This is the " + string.Join(" the ", Enumerable.Range(1, verseNumber).Reverse().Select((verse) => string.Join(" that ", nameAction[verse]))) + ".";

    public static string Recite(int startVerse, int endVerse) => string.Join("\n", Enumerable.Range(startVerse, endVerse - startVerse + 1).Select(verse => Recite(verse)));
}

File: Tracks/typescript/house/house.ts
export default class House {
  private static whatDidWhat = [
    { what: "horse and the hound and the horn", didWhat: "belonged to" },
    { what: "farmer sowing his corn", didWhat: "kept" },
    { what: "rooster that crowed in the morn", didWhat: "woke" },
    { what: "priest all shaven and shorn", didWhat: "married" },
    { what: "man all tattered and torn", didWhat: "kissed" },
    { what: "maiden all forlorn", didWhat: "milked" },
    { what: "cow with the crumpled horn", didWhat: "tossed" },
    { what: "dog", didWhat: "worried" },
    { what: "cat", didWhat: "killed" },
    { what: "rat", didWhat: "ate" },
    { what: "malt", didWhat: "lay in" },
    { what: "house that Jack built.", didWhat: "" }
  ];

  static verse = (verse: number): string[] =>
    House.whatDidWhat
      .slice(House.whatDidWhat.length - verse)
      .reduce<string[]>((verse, current, index, whatDidWhat) => {
        if (verse.length === 0) {
          verse.push(`This is the ${current.what}`);
        } else {
          verse.push(
            `that ${whatDidWhat[index - 1].didWhat} the ${current.what}`
          );
        }
        return verse;
      }, []);

  static verses = (firstVerse: number, lastVerse: number): string[] =>
    [...Array(lastVerse - firstVerse + 1)]
      .map((_, index) => index + firstVerse)
      .reduce<string[]>(
        (verses, verse) =>
          (verses.length === 0 && [...House.verse(verse)]) ||
          verses.concat(["", ...House.verse(verse)]),
        []
      );
}
